webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/reports/ReportingConfiguration.scala

```diff
@@ -65,28 +68,53 @@ final case class HeartbeatConfiguration(
     heartbeatPeriod: Int
 )
 
+object HeartbeatConfiguration {
+  implicit val codecHeartbeatConfiguration: JsonCodec[HeartbeatConfiguration] = DeriveJsonCodec.gen
+}
+
+@jsonExplicitNull
+final case class ElmAgentRunInterval(
+    overrides:   Option[Boolean], // true, false, null because used in flag
+    interval:    Int,
+    startHour:   Int,
+    startMinute: Int,
+    splayHour:   Int,
+    splayMinute: Int
+)
+object ElmAgentRunInterval    {
+  implicit val codecJsonAgentRunInterval: JsonCodec[ElmAgentRunInterval] = DeriveJsonCodec.gen
```

Comment:
I am not a fan of the `Elm` naming here, moreover I understand that the "Decoder" part is unused in prod, this case class is used to "split" the `splaytime` into `splayHour` and `splayMinute` for encoding, which is not specific to `Elm` at all, we may want to have it available via API...
And now since the decoder is provided with the declared codec, we are tempted to use it in tests... maybe we should limit the case class to the current "split splaytime and encode" use case and only provide the encoder part ?
(The visibility of the class may even be `private`, only accessible with through the `jsonWithSplay` method)

Reply:
Ok for the codec, I though I had change it already. 
For the elm part, it's really because it's the Elm semantic for JS to get a `Maybe[Boolean]`, there is no other reason to use the null case without that, our convention everywhere are to not have optionnal fields in JSON in the `None` case. 
Here, we only need it because: 
- it's used in the init part of the application
- there's not dedicated decoder for it in elm
- so it relies on Elm automatic mapping. 

I didn't want to change things in the Elm part because it's not the goal of these PR, the goal is just to remove lift-json without altering existing interface if possible - which it is, here. 
We can totally do a latter refactoring to change the elm init method of that app, and make it use a correct decoder, so that the `overrides: null` case can go away.

Reply:
OK made it clearer by moving all things specific to `AgentScheduleEditForm` into that class.  
And really `AgentScheduleEditForm` and corresponding elm app are ripped for refactoring:
- the decoding is done with lift-json parsing and is not clearly isomorph to the encoding,
- the init is bound to lift
- other strange things